### ISSUE TITLE
Add support for parallel batch processing messages

### ISSUE THREAD

**Issue by astubbs** [confluentinc/parallel-consumer#18]
Currently the interface only provides the user a way to process single messages in parallel. There are use cases however where the user may want to processes batches of messages in parallel as well.

For example, say the API they're calling out to actually supports batches of requests, for example Elastic Search batch API that can take several documents at once.

The API might look like: 
```
void pollBatch(int batchLevel, Consumer<List<ConsumerRecord<K, V>>> usersVoidConsumptionFunction);
```
So the user could request batches of 5 records, which could then be transformed into a single HTTP request. These batches of 5 messages would all be run in parallel, up to the usual concurrency limits. The same could be added to the vertx module.

Further from this, options could be added to specify requirements for the batch - for example, give me 5 messages in a batch that all have the same key, are all in the same partition etc.

**Comment by astubbs** [comment]
POC https://github.com/confluentinc/parallel-consumer/pull/19

**Comment by raphaelauv** [comment]
This feature is absolutely necessary for my use case

And since the official Kafka consumer lib is bulk I think this wrapper should support it.

**Comment by ismarslomic** [comment]
I can also confirm that this feature is highly wanted. I´m working as we speak with a use case where we need to move records from Kafka to Elasticsearch and doing this in bulks would reduce high volume of REST API calls

**Comment by raphaelauv** [comment]
this will not be part of release 1.0.0 ?

**Comment by astubbs** [comment]
@raphaelauv yes I think it should be. I'm currently working on another project atm, but hope to get back to PC 1.0 release soon.

**Comment by astubbs** [comment]
FYI ok, there’s a couple test failures still - i think they need updating. But the batching feature seems feature complete now (lol). Need to implement some test for the vertx and reactor versions, but should have this merged early next week.

**Comment by astubbs** [comment]
> this will not be part of release 1.0.0 ?

Merged
